### ISSUE TITLE
Constructor-argument for a record is also applied to all other records in the same object-tree

### ISSUE THREAD

**Issue by jk-idealo** [Nylle/JavaFixture#92]
Creating a record works just fine. Also customizing, as the following test proves:
```java
class RecordTest {
    public record TestRecord(String someString){}

    @Test
    void thisIsWorking() {
        var actual = fixture().build(TestRecord.class).with("someString", "foo").create();
        assertThat(actual.someString()).isEqualTo("foo");
    }
}
```
However, if my record has a nested record, it throws a generic `SpecimenException`:
```java
class RecordTest {
    public record TestRecord(String someString, OtherTestRecord otherTestRecord) {}
    public record OtherTestRecord(String anotherString) {}

    @Test
    void thisThrows() {
        var actual = fixture().build(TestRecord.class).with("someString", "foo").create();
        assertThat(actual.someString()).isEqualTo("foo");
    }
}
```
The exception is:
```
com.github.nylle.javafixture.SpecimenException: Cannot manufacture class namespace.RecordTest$TestRecord
	at com.github.nylle.javafixture.InstanceFactory.lambda$manufacture$5(InstanceFactory.java:77)
	at java.base/java.util.Optional.orElseThrow(Optional.java:403)
	at com.github.nylle.javafixture.InstanceFactory.manufacture(InstanceFactory.java:77)
	at com.github.nylle.javafixture.InstanceFactory.construct(InstanceFactory.java:119)
	at com.github.nylle.javafixture.InstanceFactory.construct(InstanceFactory.java:63)
	at com.github.nylle.javafixture.specimen.ObjectSpecimen.populate(ObjectSpecimen.java:74)
	at com.github.nylle.javafixture.specimen.ObjectSpecimen.create(ObjectSpecimen.java:57)
	at com.github.nylle.javafixture.SpecimenBuilder.create(SpecimenBuilder.java:33)
	at namespace.RecordTest.thisThrows(RecordTest.java:81)
       ...
```

When debugging, I found out, that the customization-field "someString" is not only applied to the constructor of `TestRecord`, but it is **also** applied to the constructor of `OtherTestRecord` where it doesn't exist. It even throws the appropriate exception:
```
com.github.nylle.javafixture.SpecimenException: Cannot customize field 'someString': Field not found in class 'namespace.RecordTest$OtherTestRecord'.
```
But this exception is being caught inside `InstanceFactory::construct` and the fallback to `InstanceFactory::manufacture` is executed, which cannot find a factory-method and throws.

**Comment by akutschera** [comment]
When I wrote [we wait until somebody complains](https://github.com/Nylle/JavaFixture/issues/87#issuecomment-1490568512) I didn't expect this to happen so quickly.

I guess the expected behaviour is that field `someString` is ignored when creating `OtherTestRecord`.

We can make this work in two ways:

1.  we only use field names that we can find (we could print out a warning). Of course this would mean that if you mis-type `somSring` it won't be set.

1. we could use each customization field exactly once (on the top-level object). Having a nested object with the same field name that you expect to customize with the same value is then unsupported.

Right now I don't want to set custom fields for nested objects. If you need that you will have to create a customized nested record and use `.with("otherTestRecord", customizedOtherTestRecord)`. This is also a workaround for this bug.

**Comment by Nylle** [comment]
> I guess the expected behaviour is that field `someString` is ignored when creating `OtherTestRecord`.

Correct. That's how it works for class customisations as well. 

> 2. we could use each customization field exactly once (on the top-level object). 

Yes. This is how it behaves for classes already.

> Having a nested object with the same field name that you expect to customize with the same value is then unsupported.

Correct, just like any object other than records (and maybe some strange immutable objects I never encountered).

**Comment by jk-idealo** [comment]
Unfortunately, the problem still occurs.

This test is **green**, when I customise the first `String` whose constructor-argument comes _before_ `OtherTestRecord`:
```java
    public record TestRecord(String string1, OtherTestRecord otherTestRecord, String string2) {}
    public record OtherTestRecord(String otherString) {}

    @Test
    void xxx() {
        var actual = fixture().build(TestRecord.class).with("string1", "foo").create();
        assertThat(actual.string1()).isEqualTo("foo");
    }
```

But if I customise the _second_ `String` whose constructor-argument comes _after_ `OtherTestRecord`, the test is **red** again:
```java
    public record TestRecord(String string1, OtherTestRecord otherTestRecord, String string2) {}
    public record OtherTestRecord(String otherString) {}

    @Test
    void xxx() {
        var actual = fixture().build(TestRecord.class).with("string2", "foo").create();
        assertThat(actual.string2()).isEqualTo("foo");
    }
```

I first thought, the problem is the order of the arguments, but funnily enough, if I remove the first `String` altogether it works again:
```java
    public record TestRecord(OtherTestRecord otherTestRecord, String string2) {}
    public record OtherTestRecord(String otherString) {}

    @Test
    void thisThrows() {
        var actual = fixture().build(TestRecord.class).with("string2", "foo").create();
        assertThat(actual.string2()).isEqualTo("foo");
    }
```

**Comment by akutschera** [comment]
Oh. I overlooked the second case. I refactored the CustomizationContext a little.
Before it was only possible to customize either "use constructor" or "ignore or customize field".
Now it should be a little easier to understand and we don't need to change the customization lists on the fly.

**Comment by Nylle** [comment]
Thank you for your help and engagement!

**Comment by jk-idealo** [comment]
Fixed in release [2.9.8](https://github.com/Nylle/JavaFixture/releases/tag/2.9.8).
